java/statestore/src/main/java/sleeper/statestore/s3/S3FileInfo.java

```diff
@@ -18,73 +18,113 @@
 
 import sleeper.core.statestore.FileInfo;
 
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
 import java.util.Objects;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public class S3FileInfo {
-    public enum FileStatus {
-        ACTIVE, READY_FOR_GARBAGE_COLLECTION
-    }
 
-    private final FileInfo fileInfo;
-    private final FileStatus status;
+    private final String filename;
+    private final List<FileInfo> internalReferences;
+    private final int externalReferenceCount;
+    private final Instant lastUpdateTime;
 
     private S3FileInfo(Builder builder) {
-        fileInfo = builder.fileInfo;
-        status = builder.status;
+        filename = builder.filename;
+        internalReferences = builder.internalReferences;
+        externalReferenceCount = builder.externalReferenceCount;
+        lastUpdateTime = builder.lastUpdateTime;
     }
 
     public static Builder builder() {
         return new Builder();
     }
 
-    public static S3FileInfo active(FileInfo fileInfo) {
-        return builder().fileInfo(fileInfo)
-                .status(FileStatus.ACTIVE).build();
+    public static List<S3FileInfo> newFiles(List<FileInfo> references, Instant updateTime) {
```

Comment:
Can you add some Javadoc to this method? And I wonder if there's a better name for the method - newFiles doesn't tell me much about what the method is for.